### PR TITLE
Refetch field snippet when field config changes

### PR DESCRIPTION
**Pull request by dani-mp**
**Resolves**: 
https://linear.app/prismic/issue/DT-2433/aauser-when-i-edit-a-link-field-i-can-see-the-code-snippet-change

### Description

Use a dynamic key based on the config of some fields to make useSWR refetch the field snippet to show. More as comments in the PR.

### Checklist

- [ ] If my changes require tests, I added them.
- [ ] If my changes affect backward compatibility, it has been discussed.
- [ ] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### How to QA [^1]

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx [hunk 1]
```diff
@@ -20,9 +20,15 @@ const Hint: React.FC<HintProps> = ({
 }) => {
   const fieldPathString = renderHintBase({ item });
 
-  // TODO: Call `swr`'s global `mutate` function when something changes to clear the cache.
```
Comment (dani-mp) [inline packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx:23]: I also tried this solution, but I didn't like it. We have at least two places where we'd have to make this call, making sure the key is the same in both places, and it wouldn't be declarative.

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx [hunk 2]
```diff
@@ -20,9 +20,15 @@ const Hint: React.FC<HintProps> = ({
...
+  const snippetCacheKey = [fieldPathString];
+  if (item.value.type === "Link") {
+    if (item.value.config?.allowText ?? false)
+      snippetCacheKey.push("allowText");
+    if (item.value.config?.repeat ?? false) snippetCacheKey.push("repeat");
```
Comment (dani-mp) [inline packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx]:
```suggestion
```

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx [hunk 3]
```diff
@@ -20,9 +20,14 @@ const Hint: React.FC<HintProps> = ({
 }) => {
   const fieldPathString = renderHintBase({ item });
 
-  // TODO: Call `swr`'s global `mutate` function when something changes to clear the cache.
+  const snippetCacheKey = [fieldPathString];
+  if (item.value.type === "Link") {
+    if (item.value.config?.allowText ?? false)
+      snippetCacheKey.push("allowText");
```
Comment (dani-mp) [inline packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx:26]: Here we'll add also the new `repeat` config so it also reacts to it.

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx [hunk 4]
```diff
@@ -20,9 +20,14 @@ const Hint: React.FC<HintProps> = ({
 }) => {
   const fieldPathString = renderHintBase({ item });
 
-  // TODO: Call `swr`'s global `mutate` function when something changes to clear the cache.
+  const snippetCacheKey = [fieldPathString];
```
Comment (dani-mp) [inline packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx:23]:
```suggestion
  const snippetCacheKey = [item.value.type];
```

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx [hunk 5]
```diff
@@ -20,9 +20,14 @@ const Hint: React.FC<HintProps> = ({
 }) => {
   const fieldPathString = renderHintBase({ item });
 
-  // TODO: Call `swr`'s global `mutate` function when something changes to clear the cache.
+  const snippetCacheKey = [item.value.type];
```
Comment (dani-mp) [inline packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx]:
```suggestion
  const snippetCacheKey: string[] = [item.value.type];
```

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx [hunk 6]
```diff
@@ -20,9 +20,14 @@ const Hint: React.FC<HintProps> = ({
 }) => {
   const fieldPathString = renderHintBase({ item });
 
-  // TODO: Call `swr`'s global `mutate` function when something changes to clear the cache.
+  const snippetCacheKey: string[] = [item.value.type];
```
Comment (dani-mp) [inline packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx]:
```suggestion
  const snippetCacheKey = [fieldPathString];
```

**Review by jomifepe — APPROVED** [review]
LGTM ✅

File: packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx (hunk 3 above)
Comment (jomifepe) [inline packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/components/Hints/index.tsx:26]: Nit 😇

```suggestion
    if (item.value.config?.allowText ?? false) {
      snippetCacheKey.push("allowText");
    }
```

**Pull request merged**